### PR TITLE
Introduce `Utils::Files`

### PR DESCRIPTION
**Pull request by jodosha**
This is a set of helpers that we were using since long time in `hanami` test suite.

With the new CLI for `hanami`, based on `hanami-cli`, we needed to extract that helpers from the test suite and promote it as production code.

Ref https://github.com/hanami/hanami/pull/808

### CODE REVIEW

File: lib/hanami/utils/files.rb [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+      # @see .mkdir
+      #
+      # @example
+      #   require "hanami/utils/files"
+      #
+      #   Hanami::Utils::Files.mkdir_p("path/to/file.rb")
+      #     # => creates the `path/to` directory, but NOT `file.rb`
+      #
+      #   # WRONG it doesn't create the last directory, check `.mkdir`
+      #   Hanami::Utils::Files.mkdir("path/to/directory")
```
Comment (AlfonsoUceda) [inline lib/hanami/utils/files.rb]: I think you need to use `mkdir_p`

File: lib/hanami/utils/files.rb [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+      #   # class App
+      #   #   configure do
+      #   #     root __dir__
+      #   #   end
+      #   # end
+      #
+      #   Hanami::Utils::Files.remove_block("app.rb", "configure")
+      #
+      #
+      #
```
Comment (AlfonsoUceda) [inline lib/hanami/utils/files.rb]: I think you can read some lines here

File: spec/unit/hanami/utils/files_spec.rb [hunk 3]
```diff
@@ -0,0 +1,868 @@
...
+
+    it "creates intermediate directories" do
+      path = root.join("path", "to", "file", "write")
+      described_class.write(path, ":)")
+
+      expect(path).to exist
+      expect(path).to have_content(":)")
+    end
+
+    it "replaces previous contentes" do
```
Comment (AlfonsoUceda) [inline spec/unit/hanami/utils/files_spec.rb]: contents?

File: spec/unit/hanami/utils/files_spec.rb [hunk 4]
```diff
@@ -0,0 +1,868 @@
...
+  end
+
+  describe ".cp" do
+    let(:source) { root.join("..", "source") }
+
+    before do
+      source.delete if source.exist?
+    end
+
+    it "creates an file with given contents" do
```
Comment (AlfonsoUceda) [inline spec/unit/hanami/utils/files_spec.rb]: a file

File: lib/hanami/utils/files.rb [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+      # @param target [String,Regexp] the target to replace
+      # @param replacement [String] the replacement
+      #
+      # @raises [Errno::ENOENT] if the path doesn't exist
+      # @raises [ArgumentError] if `target` cannot be found in `path`
+      #
+      # @see .replace_last_line
+      #
+      # @since x.x.x
+      def self.replace_line(path, target, replacement)
```
Comment (AlfonsoUceda) [inline lib/hanami/utils/files.rb]: because we have `.replace_last_line` why not rename to `replace_first_line`?

**Comment by jodosha** [comment]
Hey Alfonso, if you want, please fix/rename stuff at your wish. 😉

http://lucaguidi.com

**Comment by AlfonsoUceda** [comment]
@jodosha done!

**Pull request merged**